cosmwasm/packages/registry/src/registry.rs

```diff
@@ -19,6 +19,41 @@ pub fn hash_denom_trace(unwrapped: &str) -> String {
     format!("ibc/{}", hash.to_uppercase())
 }
 
+// When a contract is called using ibc callbacks, the addres is a combination of
+// the channel and the original sender. This function lets us compute that.
+pub fn derive_wasmhooks_sender(
+    channel: &str,
+    original_sender: &str,
+    bech32_prefix: &str,
+) -> Result<String, RegistryError> {
+    let sender = format!("{}/{}", channel, original_sender);
+
+    let mut hasher0 = Sha256::new();
+    hasher0.update("ibc-wasm-hook-intermediary".as_bytes());
+    let th = hasher0.finalize();
+
+    let mut hasher = Sha256::new();
+    hasher.reset();
+    hasher.update(th.as_slice());
+    hasher.update(sender.as_bytes());
+
+    let result = hasher.finalize();
+
+    let result_u5 = bech32::convert_bits(result.as_slice(), 8, 5, true)?;
```

Comment:
comments on what's happening here/ some reference would be nice. Eg. 
- why hashing "ibc-wasm-hook-intermediary" string
- why convert 8 to 5 bits with padding. (and can't this be just `result.to_base32()` without the `filter_map` below?)

Reply:
just a thought but I think `ibc-hook` deserves to have it's own crate for using with contract. Or maybe live in `osmosis-std`.

Reply:
yeah, this should probably live in osmosis-std. @iboss-ptk Can we work together in designing a good reusable api for it?

>  can't this be just result.to_base32()

No, but I can't remember why. I'll go through the code and add some comments for the future

Reply:
added some comments. AFAICT there is no cleaner way to do this, but happy to be proven wrong